test/built-ins/Iterator/concat/throws-typeerror-when-generator-is-running-return.js

```diff
@@ -0,0 +1,53 @@
+// Copyright (C) 2024 André Bargull. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-iterator.concat
+description: >
+  Throws a TypeError when the closure generator is already running.
+info: |
+  27.1.2.1.2 %IteratorHelperPrototype%.return ( )
+    ...
+    6. Return ? GeneratorResumeAbrupt(O, C, "Iterator Helper").
+
+  27.5.3.4 GeneratorResumeAbrupt ( generator, abruptCompletion, generatorBrand )
+    1. Let state be ? GeneratorValidate(generator, generatorBrand).
+    ...
+
+  27.5.3.2 GeneratorValidate ( generator, generatorBrand )
+    ...
+    6. If state is executing, throw a TypeError exception.
+    ...
+features: [iterator-sequencing]
+---*/
+
+let enterCount = 0;
+
+let testIterator = {
+  next() {
+    return {done: false};
+  },
+  return() {
+    enterCount++;
+    iterator.return();
+    return {done: false};
+  }
+}
+
+let iterable = {
+  [Symbol.iterator]() {
+    return testIterator;
+  }
+};
+
+let iterator = Iterator.concat(iterable);
+
+iterator.next();
+
+assert.sameValue(enterCount, 0);
+
+assert.throws(TypeError, function() {
+  iterator.return();
```

Comment:
fixed the other two issues, but now mine doesn't throw here.

Reply:
1. this line calls `%IteratorHelperPrototype%.return`
1. `%IteratorHelperPrototype%.return` step 6 calls `GeneratorResumeAbrupt`
1. `GeneratorResumeAbrupt` step 8 sets `[[GeneratorState]]` to executing
1. `GeneratorResumeAbrupt` step 10 resumes the underlying generator
1. `Iterator.concat` step 3.a.v.3.a, Yield produces the return completion from `%IteratorHelperPrototype%.return` step 5
1. `Iterator.concat` step 3.a.v.3.b.i calls `IteratorClose`
1. `IteratorClose` step 4.c calls `testIterator.return`
1. line 32 calls `%IteratorHelperPrototype%.return`
1. `%IteratorHelperPrototype%.return` step 6 calls `GeneratorResumeAbrupt`
1. `GeneratorResumeAbrupt` step 1 calls `GeneratorValidate`
1. `GeneratorValidate` step 6 throws

Reply:
https://github.com/es-shims/iterator-helpers/blob/dd80b2f1718092219e94e24afdc1737f062e3df4/aos/GeneratorResumeAbrupt.js#L34-L37 seems to be missing to implement GeneratorResumeAbrupt, step 8 where `generator.[[GeneratorState]]` is set to `executing`.

Reply:
I've pushed up some local changes I had, so now: if i comment out/remove this (incorrect) block https://github.com/es-shims/iterator-helpers/blob/temp/aos/GeneratorResumeAbrupt.js#L39-L42

then i go from 1 failing test to 9 (from `node test`). I'm pretty convinced this is a polyfill problem, though, because I'm not using generators and I've made my own quasi-state machine that surely hits some edge cases.